### ISSUE TITLE
AddressSanitizer failing in nightly builds

### ISSUE THREAD

**Issue by dstebila** [open-quantum-safe/liboqs#738]
https://app.circleci.com/pipelines/github/open-quantum-safe/liboqs/685/workflows/2712b1d5-7cfa-4e07-8a26-9eee909afee5/jobs/7147

**Comment by dstebila** [comment]
I'm unable to reproduce this locally with circleci local on either of my two machines. Is anyone else able to? @baentsch @xvzcf 

```
circleci config process .circleci/config.yml > .circleci/config.yml.expanded
circleci local execute --config=.circleci/config.yml.expanded --job ubuntu-bionic-x86_64-asan
```

**Comment by xvzcf** [comment]
I am unable to do so.

**Comment by baentsch** [comment]
Both on my regular Linux VM and OSX natively:
 =================== 381 passed, 8 skipped in 614.32 seconds ====================
so, No, sorry, can't reproduce.

On the hunch that limited resources are the problem, I also set up a small Linux VM (2 cores, 2G RAM) and retried there: It just took an eternity to run but then returned the same result: All OK.

**Comment by dstebila** [comment]
Thanks guys. So should I turn it asan off from the nightly builds? Or skip the SPHINCS schemes for asan on CircleCI?

**Comment by xvzcf** [comment]
Unless many more seemingly-hard-to-debug errors arise, we should just skip SPHINCS for now.

**Comment by dstebila** [comment]
Okay, I'll try to work on that sometime.

**Comment by dstebila** [comment]
Something mentioned on today's call made me wonder whether pytest running in parallel with asan turned on is exhausting the limited resources of CircleCI's build machine.  @xvzcf, could we come up with some CMake cleverness so that the run\_tests command doesn't add the `--numprocesses=auto` flag when `"${CIRCLE_JOB}"="ubuntu-bionic-x86_64-asan" && "x${CIRCLE_BUILD_NUM}" != "x"` (the latter condition (checking whether CIRCLE\_BUILD\_NUM is empty or not) is, as far as I can tell, a way of detecting that it is being run live on a CircleCI server, rather than with `circleci local`)?

**Comment by xvzcf** [comment]
Looking into it.

**Comment by xvzcf** [comment]
It seems you both were right @baentsch and @dstebila, [this build](https://app.circleci.com/pipelines/github/open-quantum-safe/liboqs/699/workflows/82d6220d-d149-4890-87ee-25d15e319389/jobs/7265/steps) running with `--numprocesses=1` passes. I'll open the PR tomorrow.

**Comment by dstebila** [comment]
Fixed by #743.
